### ISSUE TITLE
Permission Issues with Admin Accounts, Group/Agent Access, Console Access, Sharing and Ownerwhip

### ISSUE THREAD

**Issue by prononext** [Ylianst/MeshCentral#3857]
I noticed that there are some permission problems on admin accounts and adding group access to users.
Main reason of noticing was, that the console access on secondary admin user was lost. 

**First I want to explain how I understand the primary and first created admin account in MeshCentral:** 

1. Has full access to all ressources on the server
2. Has full access to all groups, also the ones that new users were creating
3. Has full access to all agents and clients, also the ones that new users were creating
4. Right of this first admin account cannot be altered and changed in any way it is the only full admin account on the server!

**As the situation seems to be very complex I try to explain as simple as possible and what issues I was discovering on debugging:**

1. First created admin account permissions can be altered from other admin accounts
2. First created admin account has no access to all groups which users have created
3. First created admin acount has no access to all agents and clients users have created
4. Randomly, some groups can be shared and some not, the sharing is made but the group name is not added to the list of Device Groups
5. All Agent Console access on admin accounts is lost, if they have been Full Admin one time. Can only be get by entering the specific device under the user with checkbox on Agent Consosle permissions, but also randomly not for all agents.
6. Console Access is possible only if single Device is entered in the admin with checkbox on Agent Console Permissions. Full rights on groups does not include console access, but also randomly not for all agents. 
7. If the console access if given to admin users by adding the device manually the console does not work, commands can be entered but there is no response. 
8. Console Access is possible for normal users which have full rights on groups
9. Group ownership cannot be changed by anyone, not user, not admin
10. Agent ownership cannot be changed by anyone, not user, not admin
11. Group access given to admin users cannot be removed by other admin users
12. All of these actions in the gui which are not working as expected do not give any server or gui error message. 

All these actions were tested in the gui multiple times with serveral restarts. 

Server Version: 1.0.2
System: Ubuntu 20.04
DB: MongoDB
Mode: WAN only on FQDN
The Server Config was already posted here https://github.com/Ylianst/MeshCentral/issues/3851

**Comment by Ylianst** [comment]
So, account have two types of permissions. There are server administration permissions, and device group permissions. When you are administrator on the server, you do not by default get to see or manage all device groups. For example, on [MeshCentral.com](https://meshcentral.com), I am administrator of the server but my account only have a few device groups in it, I do not see or manage any other device groups except my own.

If you want to see all device groups on the server, there is a special option for this in the config.json:

```
"manageAllDeviceGroups": "admin"
```

This will allow the "admin" account to see all device groups, you can also specify multiple accounts:

```
"manageAllDeviceGroups": [ "admin", "bob", "joe" ]
```

Note that this only works if the account is also a full administrator on the server. So, to review your points. The first administrator for the server is not a special account, any account can be promoted to full site administrator.

- Has full access to all resources on the server
- Has full access to all groups, also the ones that new users were creating
- Has full access to all agents and clients, also the ones that new users were creating

No, to all 3. First account is server administrator but does see all of the users device groups unless "manageAllDeviceGroups" is used.

- Right of this first admin account cannot be altered and changed in any way it is the only full admin account on the server!

No. There is nothing special about the first account. Any other account can be upgraded to server administrator and your can use a different administrator account to remove administrator right to the first account. If you ever loose the access to the administrator account on your server, go in a shell and run this:

```
node node_modules/meshcentral --help
```

There will be command line switches to recover a admin account. Let me know if this helps.

**Comment by prononext** [comment]
Thank you, that is a logical function. 

When activating the "manageAllDeviceGroups" the selected users have all permissions. 

**Be aware when "userNameIsEmail" is active the settings string should look like this:** 

"manageAllDeviceGroups": [ "user//user1@example.com", "user//user2@example.com" ],

I finally found out the missing console problem also, because I was wondering why the users, which I had activated to be in the "manageAllDeviceGroups" setting show only partial rights on all the groups when inside the "MyAccount" page. 

I tested a ton of options and finally found out, that if you activate "Record Sessions" inside that admin user, the permissions on all groups change from "Full Administrator" to "Partial Rights", mentioning that I also have activated server recording also. With this setting in the admin user, he also looses Agent Console access to all agents. 

I removed the "Record Session" option inside that admin user and set it inside the group and see: Console Access and Full Administrator rights were there again. 

Guess that is a minor bug to fix... Thank you Ylianst
